custom/enikshay/case_utils.py

```diff
@@ -170,6 +172,39 @@ def get_open_episode_case_from_person(domain, person_case_id):
     )
 
 
+def get_open_referral_case_from_person(domain, person_case_id):
+    episode = get_open_episode_case_from_person(domain, person_case_id)
+    case_accessor = CaseAccessors(domain)
+    reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    open_referral_cases = [
+        case for case in reverse_indexed_cases
+        if not case.closed and case.type == CASE_TYPE_REFERRAL
+    ]
+    if not open_referral_cases:
+        return None
+    if len(open_referral_cases) == 1:
+        return open_referral_cases[0]
+    else:
+        raise ENikshayException(
+            "Expected none or one open referral case for person with id: {}".format(person_case_id)
+        )
+
+
+def get_latest_trail_case_from_person(domain, person_case_id):
+    episode = get_open_episode_case_from_person(domain, person_case_id)
+    case_accessor = CaseAccessors(domain)
+    reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    trail_cases = [
+        case for case in reverse_indexed_cases
+        if case.type == CASE_TYPE_TRAIL
+    ]
+    trail_cases.sort(key=lambda c: c.opened_on)
```

Comment:
Thanks for the feedback Ben.
I'm thinking it might not be worth adding the column index. For one, I don't think this will be a super common query, as it's specific to one project, and also I think the number of extension cases that need to be sorted will be less than 100, so doesn't seem like the index would provide such a big improvement.
Also, without the ordered_by index we will still have the advantage of returning less content, and also making the subsequent `get_multiple_cases_indices` call smaller.

Another thing to consider is that the couch version of this method will have to do all the filtering and sorting in the application layer, unless we change the couch view. Do you think it's misleading to add parameters to CaseAccessor functions that won't be used at the db level?